libs/libxx/libcxx_nuttx_workaround.patch

```diff
@@ -0,0 +1,19 @@
+--- libs/libxx/libcxx/include/wchar.h	2023-12-04 15:32:42.227766906 +0100
++++ libs/libxx/libcxx/include/wchar.h	2023-12-04 15:32:36.163715150 +0100
+@@ -7,15 +7,7 @@
+ //
+ //===----------------------------------------------------------------------===//
+ 
+-#if defined(__need_wint_t) || defined(__need_mbstate_t)
```

Comment:
@xiaoxiang781216 here is the patch file. Hopefully the build passes. But i have no clue what repercussion this might have in the rest of the code. As far as cxx examples go, they seem to be functional.